CorsixTH/Lua/dialogs/resizables/adviser_history.lua

```diff
@@ -0,0 +1,181 @@
+--[[ Copyright (c) 2025 Damian "ShiroAka"
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of
+this software and associated documentation files (the "Software"), to deal in
+the Software without restriction, including without limitation the rights to
+use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+of the Software, and to permit persons to whom the Software is furnished to do
+so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE. --]]
+
+--! Adviser message history
+class "UIAdviserHistory" (UIResizable)
+
+---@type UIAdviserHistory
+local UIAdviserHistory = _G["UIAdviserHistory"]
+
+local col_bg = { red = 60, green = 174, blue = 201, }
+local col_highlight = { red = 47, green = 154, blue = 190, }
+local col_shadow = { red = 36, green = 138, blue = 158 }
+local col_delete_button = { red = 219, green = 36, blue = 36 }
+
+-- Timer, used to update the dialog only once per X ticks to reduce the burden.
+local ticks_to_skip = 5
+local tick_timer = ticks_to_skip
+
+local row_height = 40
+local header_height = 20
+local window_margin = 15
+
+function UIAdviserHistory:UIAdviserHistory(ui)
+  local app = ui.app
+
+  -- Calculate menu's height
+  local height_divisor = 4
+  local approx_rows_space = math.floor(app.config.height / height_divisor)
+  self.rows = math.ceil(approx_rows_space / row_height)
+  local rows_space = self.rows * row_height
+
+  local width = 500
```

Comment:
You may want to move this line to the other constants at lines 36-38.
In that way, all magic sizes are near each other. No need to go look through the entire file to find them then.